src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
`startDocument` & `endDocument` confuse because of sax api methods name http://www.saxproject.org/apidoc/org/xml/sax/ContentHandler.html#startDocument()